roles/base/tasks/main.yml

```diff
@@ -1,72 +1,222 @@
+  apt_repository:
+    repo: "ppa:certbot/certbot"
+  when: ansible_os_family == "Debian"
+
+- name: Download Postgres RPM GPG key (RHEL)
+  get_url:
+    url: "https://download.postgresql.org/pub/repos/yum/RPM-GPG-KEY-PGDG-{{postgres_version_no_dot}}"
+    dest: "/etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-{{postgres_version_no_dot}}"
+  when: ansible_os_family == "RedHat"
+
+- name: Add Postgres RPM GPG key (RHEL)
+  rpm_key: 
+    state: present
+    key: "https://download.postgresql.org/pub/repos/yum/RPM-GPG-KEY-PGDG-{{postgres_version_no_dot}}"
+  when: ansible_os_family == "RedHat"
+
+- name: Install Postgres repo (RHEL)
+  yum_repository:
+    name: "pgdg{{postgres_version_no_dot}}"
+    file: "pgdg-{{postgres_version_no_dot}}-redhat"
+    description: "PostgreSQL {{postgres_version}} $releasever - $basearch"
+    baseurl: "https://download.postgresql.org/pub/repos/yum/{{postgres_version}}/redhat/rhel-$releasever-$basearch"
+    gpgcheck: yes
+    gpgkey: "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-{{postgres_version_no_dot}}"
+  when: ansible_os_family == "RedHat"
+
+- name: Include Postgres binaries in PATH (RHEL)
+  lineinfile:
+    dest: /etc/profile.d/postgres.sh
+    line: "export PATH=$PATH:/usr/pgsql-{{postgres_version}}/bin"
+    state: present
+    create: yes
+    owner: root
+    group: root
+    mode: 0644
+  when: ansible_os_family == "RedHat"
+
+- name: Add Postgres APT GPG key (Ubuntu)
+  apt_key: 
+    state: present
+    url: "https://www.postgresql.org/media/keys/ACCC4CF8.asc"
+  when: ansible_os_family == "Debian"
+
+- name: Install Postgres repo (Ubuntu)
+  apt_repository:
+    repo: "deb http://apt.postgresql.org/pub/repos/apt/ {{ansible_distribution_release}}-pgdg main"
+    state: present
+  when: ansible_os_family == "Debian"
+
+- name: Install certbot
+  package:
+    name: "{{certbot_package}}"
+    update_cache: yes
+
+- name: Generate certbot cert (PRD)
+  command: "{{certbot_package}} certonly -n --keep --standalone -d {{ansible_fqdn}}  --agree-tos --email {{certbot_username}}"
+  args:
+    creates: "/etc/letsencrypt/live/{{ansible_fqdn}}"
+  when: oval_environment == "PRD"
+
+- name: Generate certbot cert (STAGING)
+  command: "{{certbot_package}} certonly --staging -n --keep --standalone -d {{ansible_fqdn}}  --agree-tos --email {{certbot_username}}"
+  args:
+    creates: "/etc/letsencrypt/live/{{ansible_fqdn}}"
+  when: oval_environment != "PRD"
+
+- name: Create certbot hooks dir
+  file: 
+    path: /etc/letsencrypt/hooks
+    state: directory
+    mode: 0755
+    owner: root
+    group: root
+
+- name: Install certbot cronjob (PRD)
+  cron:
+    name: "certbot cert renewal"
+    special_time: daily
+    cron_file: certbot
+    user: root
+    state: present
+    job: "/usr/bin/{{certbot_package}} renew -n --keep --standalone --pre-hook /etc/letsencrypt/hooks/pre.sh --post-hook /etc/letsencrypt/hooks/post.sh"
+  when: oval_environment == "PRD"
+
+- name: Install certbot cronjob (STAGING)
```

Comment:
> The cron job for staging is the same as for prod, right?

I think that's a typo on my part, the STAGING renew cronjob should use `--staging` I _think_.

Lemme verify if `--staging` is needed for the renew process. If it is, I'll fix this.

If it isn't, I'll remove the extra ansible task.